src/Stratis.Bitcoin.Features.Consensus/ConsensusLoop.cs

```diff
@@ -459,14 +454,7 @@ public void ValidateBlock(RuleContext context)
         {
             this.logger.LogTrace("()");
 
-            await this.consensusRules.ValidateAndExecuteAsync(context);
-
-            // Validate the UTXO set is correctly spent.
-            this.logger.LogTrace("Executing block.");
-            using (new StopwatchDisposable(o => this.Validator.PerformanceCounter.AddBlockProcessingTime(o)))
-            {
-                this.Validator.ExecuteBlock(context);
-            }
+            await this.ConsensusRules.ValidateAndExecuteAsync(context);
 
             // Persist the changes to the coinview. This will likely only be stored in memory,
             // unless the coinview treashold is reached.
```

Comment:
I suppose this should go in its own rule as well (or at the end of the CoinViewRule)
` await this.UTXOSet.SaveChangesAsync` we can wait with this for another PR.

Reply:
ok

Reply:
created issue https://github.com/stratisproject/StratisBitcoinFullNode/issues/1438